File: AD-HOC/2443/main.rb
# frozen_string_literal:true

def print_quotient(a, b, c, d)
  numerator = a * d + b * c
  denominator = b * d
  gcd = numerator.gcd(denominator)
  puts "#{numerator / gcd} #{denominator / gcd}"
end

a, b, c, d = gets.split.map(&:to_i)
print_quotient(a, b, c, d)

File: AD-HOC/2443/main.py
import math


def print_quotient(a, b, c, d):
    numerator = a * d + b * c
    denominator = b * d
    gcd = math.gcd(numerator, denominator)
    print(f'{numerator // gcd} {denominator // gcd}')


a, b, c, d = map(int, input().split())
print_quotient(a, b, c, d)
